File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left =0;
        int right =0;
        int maxLength =0;
        Set<Character> unique = new HashSet<>();

        while (right < s.length()){
            if(!unique.contains(s.charAt(right))){
                unique.add(s.charAt(right));
                maxLength = Math.max(maxLength,right-left+1);
                right++;
            }else{
                unique.remove(s.charAt(left));
                left++;
            }

        }
        return maxLength;

    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        freq = defaultdict(int)
        left=0
        ans =0
        maxFreq=0           
        for right in range(len(s)):
            freq[s[right]]+=1
            maxFreq = max(maxFreq,freq[s[right]])
            while right-left+1 - maxFreq > k:
                freq[s[left]]-=1
                left+=1
            ans = max(ans,right-left+1)
        return ans